### PR TITLE
Regenerate code, add e2e test for failing scenario

### PR DESCRIPTION
**Pull request by echen-98**
Description of changes:

- Regenerate code after upstream code-generator changes
- Bump runtime version to 0.2.0
- Fix failing unit tests
- Improve unit test diff reporting for failing cases
- Add e2e test covering failing scenarios relevant to #22 

This PR will unblock #22. I was considering making the changes and adding them onto this PR but thought it might be more convenient to separate the many newly-generated lines from the actual change.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/e2e/tests/test_replicationgroup.py [hunk 1]
```diff
@@ -122,6 +131,29 @@ def rg_cmd_fromsnapshot(bootstrap_resources, make_rg_name, make_replication_grou
...
+        "RG_ID": make_rg_name("rg-cmd-update"),
+        "ENGINE_VERSION": "5.0.0",
+        "NUM_NODE_GROUPS": "1",
+        "REPLICAS_PER_NODE_GROUP": "1"
+    }
+
+
+@pytest.fixture(scope="module")
+def rg_cmd_update(rg_cmd_update_input, make_replication_group, rg_deletion_waiter):
+    input_dict = rg_cmd_update_input
```
Comment (RedbackThomson) [inline test/e2e/tests/test_replicationgroup.py:146]: What's the purpose of this temporary variable?

File: test/e2e/tests/test_replicationgroup.py [hunk 2]
```diff
@@ -133,6 +165,52 @@ def test_rg_cmd_fromsnapshot(self, rg_cmd_fromsnapshot):
         (reference, _) = rg_cmd_fromsnapshot
         assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "True", wait_periods=30)
 
+    # test update behavior of controller; this test can be changed to include multiple chained updates
+    def test_rg_cmd_update(self, rg_cmd_update_input, rg_cmd_update):
+        (reference, _) = rg_cmd_update
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "True", wait_periods=30)
```
Comment (RedbackThomson) [inline test/e2e/tests/test_replicationgroup.py:171]: TODO (@RedbackThomson): Move constants like "ACK.ResourceSynced" into consts in the `acktest` package.

**Review by RedbackThomson — APPROVED** [review]

**Comment by RedbackThomson** [comment]
@echen-98 Would you mind splitting up the autogenerated code into a different PR for next time? It was very difficult to find the diff that was up for review compared to the sheer quantity of generated code. Otherwise just left one inline comment :)

**Review by jaypipes — APPROVED** [review]
As @RedbackThomson noted, would be easier in future to separate out the "regenerate code" parts into a different PR :) But, I'm good with this. Thanks @echen-98!

File: test/e2e/tests/test_replicationgroup.py (hunk 1 above)
Comment (echen-98) [reply test/e2e/tests/test_replicationgroup.py:146]: I guess it just makes the below lines slightly more readable, but it's not at all necessary

**Comment by echen-98** [comment]
@RedbackThomson @jaypipes my bad, will make sure to do that next time. Thanks!

**Pull request merged**